### PR TITLE
[CORL-2766] protect common email domains from domain moderation

### PR DESCRIPTION
**Pull request by marcushaddon**
## What does this PR do?
This PR creates a hardcoded list of email domains considered to common to be moderated domain-wide.

## These changes will impact:

- [x] commenters
- [x] moderators
- [x] admins
- [ ] developers

## What changes to the GraphQL/Database Schema does this PR introduce?
None.

## Does this PR introduce any new environment variables or feature flags? 
No.

## If any indexes were added, were they added to `INDEXES.md`?
n/a

## How do I test this PR?
1. As an org mod or admin attempt to ban any of the following domains, either through the checkbox presented while managing a user's ban status, or under Configure > Moderation > Email Domain
voxmedia.com
gmail.com
gmail.co.uk
aol.com
att.net
duck.com
verizon.net
yahoo.com
yahoo.co.nz
yahoo.com.au
yahoo.co.za
xtra.co.nz
btinternet.com
yahoo.se
comcast.net
outlook.com
icloud.com
rr.com
hotmail.com
hotmail.co.uk
hotmail.es
hotmail.fr
msn.com
me.com
live.co.uk
live.com
live.co.za
rogers.com
global.co.za
ncacoral.com.au
proton.me
westnet.com.au
laposte.net
free.fr
orange.fr
outlook.fr
yahoo.fr
online.no
live.no
yahoo.no
hotmail.no
2. Observe that the domain is not moderated (in the configure view, you should see a detailed error message. In the user ban status modal, you will not see any indication that the email domain ban failed, which is a preexisting behavior and is out of the scope of this ticket).
 
## How do we deploy this PR?
No special considerations should be necessary.

### CODE REVIEW

File: src/core/server/services/tenant/tenant.ts [hunk 1]
```diff
@@ -41,6 +42,8 @@ import {
   GQLUSER_ROLE,
 } from "coral-server/graph/schema/__generated__/types";
 
+import { PROTECTED_EMAIL_DOMAINS } from "coral-common/constants";
```
Comment (kabeaty) [inline src/core/server/services/tenant/tenant.ts]: Can you move these imports up?

**Comment by nick-funk** [comment]
How do we feel about adding `fastmail.com` to this list? I only ask as I see we added `proton.me` to this list already.

**Review by nick-funk — APPROVED** [review]
Works as labelled! Ship it!

**Review by kabeaty — CHANGES_REQUESTED** [review]
Looks good overall! Left a small comment. Also, I'm seeing an error thrown when testing out using the user ban status modal to apply a common email ban that I think we will need to handle? And are you able to add a quick test for these common email domains not being able to be banned, as well?

File: src/core/server/services/tenant/tenant.ts (hunk 1 above)
Comment (marcushaddon) [reply src/core/server/services/tenant/tenant.ts]: ✅

File: src/core/server/services/tenant/tenant.ts [hunk 2]
```diff
@@ -41,6 +43,7 @@ import {
   GQLUSER_ROLE,
 } from "coral-server/graph/schema/__generated__/types";
 
+import { ERROR_CODES } from "coral-common/errors";
```
Comment (kabeaty) [inline src/core/server/services/tenant/tenant.ts]: one more import to move up :)

**Review by kabeaty — APPROVED** [review]
Looks good, thanks for the updates! Just one more small comment, but approving!

File: src/core/server/services/tenant/tenant.ts (hunk 2 above)
Comment (marcushaddon) [reply src/core/server/services/tenant/tenant.ts]: ✅

**Pull request merged**